inst/examples/multiTrackApp/server.R

```diff
@@ -0,0 +1,335 @@
+function(input, output, session) {
+  all_reactive_values <- reactiveValues(
+    single_composed_track = NULL,
+    single_composed_views = NULL,
+    single_track = NULL,
+    composed_track = NULL,
+    composed_views = NULL,
+    track1 = NULL,
+    track2 = NULL,
+    track3 = NULL,
+    track4 = NULL
+  )
+
+  observeEvent(input$reset_rule_mark, {
+    zoom_to_extent(
+      component_id = "component_1",
+      view_id = "track1",
+      duration = 2000
+    )
+  })
+
+  observeEvent(input$layout_single, {
+
+    all_reactive_values$single_track <- add_single_track(
```

Comment:
What happens if we extract all of this code block building the JSON to outside of the server function? and leave only the renderUI part?
I mean, I cannot see why we need this code inside this block (This should be within a reactive context somehow?)

Reply:
No, you are right. The idea was to add more inputs that control the plot. Here only the layout is changed but I started off with the idea that I would have inputs for the data, the visual channels and so on. This is also the reason why the above var was a reactive one.
But I did not have time to add the inputs so it remained static inside reactive context. I will change it.

Reply:
Made changes. PTAL. Converting this to a class is WIP.